### PR TITLE
update the unit tests to run using bazel.

### PR DESCRIPTION
**Pull request by vamsikrishna-siddu**
**What this PR does / why we need it**:
This pr updates the existing s390x kubevirt unit tests to run using bazel. The builder image is maintained in the IBM Cloud container Registry and the required credentials to access the image is provided to the prow job by this new label preset-kubevirt-icr-credential which will pull the username and password from a Kubernetes secret.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
     None.

```

### CODE REVIEW

File: github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml [hunk 1]
```diff
@@ -616,6 +616,22 @@ presets:
   volumeMounts:
   - name: bazeluserroot
     mountPath: /tmp/cache/bazel
+- labels:
+    preset-kubevirt-icr-credential: "true"
```
Comment (brianmcarey) [inline github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml:620]: probably worth pointing out that this is only available on the s390x bare-metal cluster
```suggestion
    preset-kubevirt-s390x-icr-credential: "true"
```

I don't see this preset being used by any other lanes outside of s390x

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 2]
```diff
@@ -425,17 +425,24 @@ presubmits:
...
+        value: "icr.io/kubevirt/builder:2409300832-cb0dc925db-s390x"
       - command:
         - /usr/local/bin/runner.sh
         - /bin/sh
         - -c
-        - make go-test
+        - >
+        cat $ICR_PASSWORD | podman login icr.io --username $(cat $ICR_USER) --password-stdin=true &&
+        make test
         image: quay.io/kubevirtci/golang:v20240923-918320b
```
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:446]: There is an issue with the formatting here.

File: github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml (hunk 1 above)
Comment (vamsikrishna-siddu) [reply github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml]: done

**Comment by vamsikrishna-siddu** [comment]
> It all looks ok to me - just one question I thought that we were happy with using `make go-test` for the unit tests on s390x so why is there a need for this change?

Hi @brianmcarey  Yes, you are Correct. Actually, I have raised this PR to verify whether the above code changes are able to pull the builder image from icr.io or not. Now after this job got successful, I got conformation that the changes are working.  So, I will close this pr and raise a new PR that adds the `preset-kubevirt-s390x-icr-credential` label.  We can later use this label for the e2e tests when the k8s-provider is ready.  As we decided earlier, we will stick with running the unit tests using make go-test for s390x. Apologies for the confusion I have created.

**Pull request closed without merging**